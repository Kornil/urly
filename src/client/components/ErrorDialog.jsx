import React from "react";
import PropTypes from "prop-types";

const ErrorDialog = ({ error }) => (
  <p>
    There was an error fetching your shortlinks, please refresh the page:{" "}
    {error}
  </p>
);

ErrorDialog.propTypes = {
  error: PropTypes.string.isRequired
};

export default ErrorDialog;
