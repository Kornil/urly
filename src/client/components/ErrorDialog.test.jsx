import { shallow } from "enzyme";
import React from "react";

import ErrorDialog from "./ErrorDialog";

it("it renders", () => {
  const wrapper = shallow(<ErrorDialog error="oh no!" />);
  expect(wrapper).toMatchSnapshot();
});
