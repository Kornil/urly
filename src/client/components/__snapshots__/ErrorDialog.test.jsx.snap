// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorDialog
    error="oh no!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "There was an error fetching your shortlinks, please refresh the page:",
        " ",
        "oh no!",
      ],
    },
    "ref": null,
    "rendered": Array [
      "There was an error fetching your shortlinks, please refresh the page:",
      " ",
      "oh no!",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "There was an error fetching your shortlinks, please refresh the page:",
          " ",
          "oh no!",
        ],
      },
      "ref": null,
      "rendered": Array [
        "There was an error fetching your shortlinks, please refresh the page:",
        " ",
        "oh no!",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
